Various character replacements per README.md

DBAWiseMan opened the pull request:
Replaced smart-quotes and em-dashes with the agreed content from the README.
Found a new bug in page-converter.js where the passed parameter wasn't referenced, so fixed that too so I could be sure my single quote replacements in converter.html didn't break anything.

TheGiddyLimit commented:
Great stuff, one small change - could you remove any `"languages": "\u2014" ` entries from bestairy JSON(s)? If the item is missing, it should render the dash automatically.

DBAWiseMan commented:
Will replace all those "languages": "\u2014" with "languages": "" in bestiary JSON. Currently working on other data clean-up in that file, e.g. Replaced yuan-ti with Yuan-ti & yuan-ti with yuan-ti; fixed Nereid's Skills, Chwinga's AC, Snarla's Type/Alignment, Geist's ability scores (and there are more I haven't got to yet).
